packages/frontend/src/common/hooks/useValidatedTimeInput.ts

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react";
+import { toNumber } from "lodash";
+
+interface useValidatedTimeInputOptions {
+  initialValue?: number;
+  formatLength?: number;
+  minValue?: number;
+  maxValue?: number;
+}
+export function useValidatedTimeInput({
+  initialValue,
+  formatLength,
+  maxValue = Number.MAX_VALUE,
+  minValue = Number.MIN_VALUE,
+}: useValidatedTimeInputOptions | undefined = {}) {
+  const [value, setValue] = useState(initialValue ?? 0);
+  const formattedValue = addLeadingZeros(value);
+  const [isError, setIsError] = useState(false);
+
+  function addLeadingZeros(value: number) {
+    if (formatLength) {
+      return value.toString().padStart(formatLength, "0");
+    }
+    return value.toString();
+  }
+  function incrementTime(change: number) {
+    changeValue(value + change);
+  }
+  function decrementTime(change: number) {
+    changeValue(value - change);
+  }
+  function isValid() {
+    return value >= minValue && value <= maxValue;
+  }
+  function changeValue(value: number) {
+    setValue(value);
+    setIsError(!isValid());
```

Comment:
We set thr value herr and isValid uses value to determine if it's valid or not. However, when setting a new value, the new value will not be used until the next rerender cycle. It makes sense to pass a parameter here to ensure that the correct value is validated here.
I'd recomment to just have a variable isValid calculated on each rerender that also represents whether or not there is an error. An error state is therefore not needed.
Remember: we dont want so sync state that depends on eachother and isError depends on the value.